File: api-node/src/models/Member.ts
import { Schema, model, Document, Types } from "mongoose";
import { MemberRole } from "../types/enums";

export interface IMember extends Document {
  orgId: Types.ObjectId;
  userId: Types.ObjectId;
  role: MemberRole;
  invitedBy?: Types.ObjectId;
  invitedAt?: Date;
  acceptedAt?: Date;
  createdAt: Date;
  updatedAt: Date;
}

const memberSchema = new Schema<IMember>(
  {
    orgId: {
      type: Schema.Types.ObjectId,
      ref: "Organization",
      required: true,
      index: true,
    },
    userId: {
      type: Schema.Types.ObjectId,
      ref: "User",
      required: true,
      index: true,
    },
    role: {
      type: String,
      enum: Object.values(MemberRole),
      default: MemberRole.VIEWER,
    },
    invitedBy: { type: Schema.Types.ObjectId, ref: "User" },
    invitedAt: { type: Date },
    acceptedAt: { type: Date },
  },
  {
    timestamps: true,
  }
);

memberSchema.index({ orgId: 1, userId: 1 }, { unique: true });

export const Member = model<IMember>("Member", memberSchema);

File: worker-python/src/models/Member.py
from datetime import datetime
from beanie import Document, Link
from pydantic import Field
from .Organization import Organization
from .User import User

class Member(Document):
    org_id: Link[Organization] = Field(..., alias="orgId")
    user_id: Link[User] = Field(..., alias="userId")
    role: str = "viewer"  # owner, maintainer, reviewer, viewer
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "members"
        indexes = [
            "orgId",
            "userId",
        ]
